From 975c3e8be136fc459688c5d5e9daf272e6092af0 Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Fri, 28 Aug 2015 06:18:01 +0000
Subject: [PATCH 7/8] ARM: dts: am437x: add DT node for gpu

Add dt node to enable SGX PowerVR driver in AM437x base dtsi file.

Signed-off-by: Darren Etheridge <detheridge@ti.com>
Signed-off-by: Eric Ruei <e-ruei1@ti.com>
---
 arch/arm/boot/dts/am4372.dtsi        | 8 ++++++++
 arch/arm/boot/dts/am437x-gp-evm.dts  | 4 ++++
 arch/arm/boot/dts/am437x-idk-evm.dts | 4 ++++
 arch/arm/boot/dts/am437x-sk-evm.dts  | 4 ++++
 arch/arm/boot/dts/am43x-epos-evm.dts | 4 ++++
 5 files changed, 24 insertions(+)
 mode change 100644 => 100755 arch/arm/boot/dts/am4372.dtsi
 mode change 100644 => 100755 arch/arm/boot/dts/am437x-gp-evm.dts
 mode change 100644 => 100755 arch/arm/boot/dts/am437x-idk-evm.dts
 mode change 100644 => 100755 arch/arm/boot/dts/am437x-sk-evm.dts
 mode change 100644 => 100755 arch/arm/boot/dts/am43x-epos-evm.dts

diff --git a/arch/arm/boot/dts/am4372.dtsi b/arch/arm/boot/dts/am4372.dtsi
old mode 100644
new mode 100755
index 72009c2..4fc40b2
--- a/arch/arm/boot/dts/am4372.dtsi
+++ b/arch/arm/boot/dts/am4372.dtsi
@@ -789,6 +789,14 @@
 			dma-names = "tx", "rx";
 		};
 
+		sgx: sgx@0x56000000 {
+			compatible = "ti,am437x-sgx530", "img,sgx530";
+			ti,hwmods = "gfx";
+			reg = <0x56000000 0x1000000>;
+			interrupts = <GIC_SPI 37 IRQ_TYPE_LEVEL_HIGH>;
+			status = "disabled";
+		};
+
 		rng: rng@48310000 {
 			compatible = "ti,omap4-rng";
 			ti,hwmods = "rng";
diff --git a/arch/arm/boot/dts/am437x-gp-evm.dts b/arch/arm/boot/dts/am437x-gp-evm.dts
old mode 100644
new mode 100755
index 5a26668..bbd3f84
--- a/arch/arm/boot/dts/am437x-gp-evm.dts
+++ b/arch/arm/boot/dts/am437x-gp-evm.dts
@@ -1241,3 +1241,7 @@
 	status = "okay";
 	ext-clk-src;
 };
+
+&sgx {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/am437x-idk-evm.dts b/arch/arm/boot/dts/am437x-idk-evm.dts
old mode 100644
new mode 100755
index f7d8656..ea017df
--- a/arch/arm/boot/dts/am437x-idk-evm.dts
+++ b/arch/arm/boot/dts/am437x-idk-evm.dts
@@ -405,3 +405,7 @@
 &cpu {
 	cpu0-supply = <&tps>;
 };
+
+&sgx {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/am437x-sk-evm.dts b/arch/arm/boot/dts/am437x-sk-evm.dts
old mode 100644
new mode 100755
index 5506e91..55a1263
--- a/arch/arm/boot/dts/am437x-sk-evm.dts
+++ b/arch/arm/boot/dts/am437x-sk-evm.dts
@@ -758,3 +758,7 @@
 &wkup_m3_ipc {
 	ti,scale-data-fw = "am43x-evm-scale-data.bin";
 };
+
+&sgx {
+	status = "okay";
+};
diff --git a/arch/arm/boot/dts/am43x-epos-evm.dts b/arch/arm/boot/dts/am43x-epos-evm.dts
old mode 100644
new mode 100755
index 92515b3..c9e3dbb
--- a/arch/arm/boot/dts/am43x-epos-evm.dts
+++ b/arch/arm/boot/dts/am43x-epos-evm.dts
@@ -869,3 +869,7 @@
 &cpu {
 	cpu0-supply = <&dcdc2>;
 };
+
+&sgx {
+	status = "okay";
+};
-- 
2.2.0

